### PR TITLE
Calculating fibonacci numbers concurrently in python

### LINKED ISSUE #6701: Calculating Fibonacci Numbers Concurrently in Python.
### Topic Suggestion
Calculating Fibonacci Numbers Concurrently in Python. 

#### Proposed title of article
 Calculating Fibonacci Numbers Concurrently in Python.

#### Proposed article introduction
This article will teach you how to calculate Fibonacci numbers concurrently in Python. Fibonacci numbers are a sequence of numbers that is the sum of the two preceding numbers plus the next two numbers. The first few numbers in the sequence are 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765, 10946 and so on. This article will also show how to use a Python function to calculate Fibonacci numbers of all sizes in a functional programming style and then demonstrate how to transform the calculation into a recursive method.

#### Key takeaways 
- What is Fibonacci numbers?
- How to calculate Fibonacci numbers in Python.
- How to calculate Fibonacci numbers concurrently in Python.
- Demonstrating how to transform the calculation into a recursive method.

### PR DESCRIPTION
**Pull request by samkanoi**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6701

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by louisefindlay23** [comment]
Plagiarism check complete

**Comment by BrianaNzivu** [comment]
Hey @samkanoi 👋 
Welcome to Section -EngEd.
Happy to have you here.

I can't see your author info. Please got through the uploading.md file again for instructions on how to put up your own page.

Regards.

File: content/articles/calculating-fibonacci-numbers-concurrently-in-Python./index.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+images:
+
+  - url: /engineering-education/calculating-fibonacci-numbers-concurrently-in-Python./hero.jpg
+    alt: Calculating Fibonacci Numbers Concurrently in Python with examples
+---
+What is so special about Fibonacci numbers? They are numbers used to calculate other mathematical problems. They also have magnetic properties that do not apply to other types of numbers.
+<!--more-->
+One of these properties is recursion used to calculate Fibonacci numbers concurrently in Python using the fastest available algorithms today. What does that mean? You can find the nth Fibonacci number using far fewer computational steps than you think! This article will cover how to calculate Fibonacci numbers concurrently in Python.
+
+### Table of contents
```
Comment (BrianaNzivu) [inline content/articles/calculating-fibonacci-numbers-concurrently-in-Python./index.md:21]: You've not linked the table of contents correctly. 
It should be done like this ;
[Prerequisites](#prerequisites)
Look at the raw format of this comment to see how.

File: content/articles/calculating-fibonacci-numbers-concurrently-in-Python./index.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
...ly if we know how many Fibonacci numbers we calculate and their values. For example, consider a program that calculates the first 5 Fibonacci numbers. There will be no more than five calls to function fib(n) since n=0..4, so we can run fib(0), fib(1), fib(2), fib(3), and fib(4) in parallel. The following example shows you how to do it in Python #!/user/bin/env python # -*- coding: utf-8 -*-  Calculating Fibonacci numbers concurrently  import time def fib(n): global count count += 1 return count print Calculating Fibonacci numbers concurrently start = time.time() for i in range(5): print i, fib(i) end = time.time() print Time took: end - start print Number of calls: count Output 0 1 1 2 3 2 3 4 5 3 5 8 13 4 8 21 34 5 13 34 55 Time taken: 0.0006 Number of calls: 6 . This output shows us that running all these functions concurrently took only 0.0006 seconds, and the total number of calls was six, which means that all functions were called at least once by the main thread. I am using range(5) instead of explicitly writing out every number because they would take up too much space. When I want to iterate over multiple numbers, it’s better to use range instead of writing out every value.
```
Comment (BrianaNzivu) [inline content/articles/calculating-fibonacci-numbers-concurrently-in-Python./index.md:36]: All code should be in code blocks i.e
```python
insert your code here
```
Please re-organize all the codes into code blocks.

Comment (BrianaNzivu) [inline content/articles/calculating-fibonacci-numbers-concurrently-in-Python./index.md:36]: I want you to structure your article well. 
Your article is more of a how-to guide. Check out this [structure](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md) .

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]
Hey @samkanoi 👋 

Just a few corrections.

Ping me when done.

**Comment by BrianaNzivu** [comment]
Hey @samkanoi 👋 
Just checking up on you. Do you need some help?

Regards.

**Pull request closed without merging**